casper/src/test/scala/coop/rchain/casper/merging/MergingCases.scala

```diff
@@ -0,0 +1,103 @@
+package coop.rchain.casper.merging
+
+import cats.syntax.all._
+import cats.effect.{Concurrent, Resource}
+import coop.rchain.casper.PrettyPrinter
+import coop.rchain.casper.util.rholang.costacc.CloseBlockDeploy
+import coop.rchain.casper.util.{ConstructDeploy, GenesisBuilder}
+import coop.rchain.casper.util.rholang.{Resources, RuntimeManager, SystemDeployUtil}
+import coop.rchain.metrics.Metrics
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.Validator.Validator
+import coop.rchain.p2p.EffectsTestInstances.LogicalTime
+import coop.rchain.rholang.interpreter.SystemProcesses.BlockData
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import coop.rchain.rspace.merger.{EventLogIndex, MergingLogic}
+import coop.rchain.rspace.merger.MergingLogic.computeRelatedSets
+import coop.rchain.shared.{EventLog, Log, Time}
+import coop.rchain.shared.scalatestcontrib.effectTest
+import monix.eval.Task
+import org.scalatest.{FlatSpec, Matchers}
+import monix.execution.Scheduler.Implicits.global
+
+class MergingCases extends FlatSpec with Matchers {
+
+  val genesisContext             = GenesisBuilder.buildGenesis(validatorsNum = 5)
+  val genesis                    = genesisContext.genesisBlock
+  implicit val logEff            = Log.log[Task]
+  implicit val timeF: Time[Task] = new LogicalTime[Task]
+
+  val runtimeManagerResource: Resource[Task, RuntimeManager[Task]] = for {
+    dirs <- Resources.copyStorage[Task](genesisContext.storageDirectory)
+    kvm  <- Resource.liftF(Resources.mkTestRNodeStoreManager[Task](dirs.storageDir))
+    rm   <- Resource.liftF(Resources.mkRuntimeManagerAt[Task](kvm))
+  } yield rm
+
+  /**
+    * Two deploys inside single state transition are using the same PVV for precharge and refund.
```

Comment:
We really need a wiki page for terminology because it takes a quite a long time for me to take PVV as perValidatorVault.